src/pybind/mgr/cephadm/serve.py

```diff
@@ -504,6 +509,11 @@ def _check_for_strays(self) -> None:
                         # and don't have a way to check if the daemon is part of iscsi service
                         # we assume that all tcmu-runner daemons are managed by cephadm
                         managed.append(name)
+                    # Don't mark daemons we just removed in the last minute as stray.
+                    # It may take some time for the mgr to become aware the daemon
+                    # had been removed.
+                    if name in self.mgr.recently_removed_daemons:
```

Comment:
this is a good start, but it turns out this problem also can present itself on daemon creation, specifically with OSDs.
https://pulpito.ceph.com/adking-2024-04-22_14:50:52-orch:cephadm-wip-adk-testing-2024-04-22-0809-distro-default-smithi/7668342/
https://pulpito.ceph.com/adking-2024-04-22_14:50:52-orch:cephadm-wip-adk-testing-2024-04-22-0809-distro-default-smithi/7668323

At least all the ones that were happening to nfs as it was removed seem to have cleared up.

Reply:
made this a more generic `recently_altered_daemons` and it now also handles daemons deployed in the last 60 seconds